Add CLIENT_AS_PRINCIPAL instructions to OAuth for Okta service app guide (#4315)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/implement-oauth-for-okta-serviceapp/main/index.md

As an Okta admin, make a `POST /oauth2/v1/clients/${yourServiceAppId}/roles` request to your org with the following required parameters to assign an admin role:

| Parameter |  Description/Value   |
| --------- |  ------------- |
| `yourServiceAppId`  |  Specify the `client_id` value from the previous response when the service app was created. In the following role assignment example, the `${yourServiceAppId}` variable name is used instead of `client_id`.|
| `type`  |  Specify the admin role to assign to the service app. |

See [Assign a Role to a client application](/docs/reference/api/roles/#assign-a-role-to-a-client-application) in the Role Assignment API reference.

> #### Request example

Reviewer: This request example is for a standard admin.
For a custom admin, two more fields are required. Do we want to add a separate example for custom admin?

Author: Good idea. I'll add it. Thanks!

```bash
curl -i -X POST \
  'https://subdomain.okta.com/oauth2/v1/clients/${yourServiceAppId}/roles' \
  -H 'Authorization: YOUR_API_KEY_HERE' \
  -H 'Content-Type: application/json' \
  -d '{
    "type": "${type}"
  }'
```

> > **Note:** The admin roles determine which resources the admin can perform the actions on (such as a specific group of users or a specific set of apps), while scopes determine the action that the admin can perform (such as manage users, read apps, and so on). Therefore, the admin roles need to have enough permissions for the scopes provided.

Reviewer: Thanks for adding this note!

5. Optional. Click the **Application rate limits** tab to adjust the rate-limit capacity percentage for this service application. By default, each new application sets this percentage at 50%.

### Assign admin roles to the OAuth 2.0 service app

> > **Note:** These instructions are for Production orgs with the **Assign admin roles to public client apps** feature enabled as well as Preview orgs.<br><br>

Author: @alimohammadi-okta We don't mention the feature flag names explicitly in the devdocs. We use the name as it appears on the Features page in the UI.

> If you have a Production org and want to turn on the **Assign admin roles to public client apps** feature, see [Manage Early Access and Beta features](https://help.okta.com/okta_help.htm?id=ext_Manage_Early_Access_features).

Assign admin roles for every OAuth 2.0 service app that you create. Service apps with assigned admin roles are constrained to the permissions and resources that are included in the role. This improves security for an org since it ensures that service apps only have access to the resources that are needed to perform their tasks. You can assign the [standard admin roles](https://help.okta.com/okta_help.htm?type=oie&id=ext-administrators-admin-comparison) or a [custom admin role](https://help.okta.com/okta_help.htm?type=oie&id=ext-about-creating-custom-admin-roles) with permissions to specific resource sets.

> <!-- Check with Vasi about what roles, permissions, and resource sets are recommended -->

Author: @vasishtajayanti-okta What did we agree on as acceptable recommended roles, permissions, and resource sets?

> If you have a Production org and want to turn on the **Assign admin roles to public client apps** feature, see [Manage Early Access and Beta features](https://help.okta.com/okta_help.htm?id=ext_Manage_Early_Access_features).

Assign admin roles for every OAuth 2.0 service app that you create. Service apps with assigned admin roles are constrained to the permissions and resources that are included in the role. This improves security for an org since it ensures that service apps only have access to the resources that are needed to perform their tasks. You can assign the [standard admin roles](https://help.okta.com/okta_help.htm?type=oie&id=ext-administrators-admin-comparison) or a [custom admin role](https://help.okta.com/okta_help.htm?type=oie&id=ext-about-creating-custom-admin-roles) with permissions to specific resource sets.

> As an Okta admin, make a `POST /oauth2/v1/clients/${yourServiceAppId}/roles` request to your org with the following required parameters to assign an admin role:

Reviewer 2: This question is for @vasishtajayanti-okta : can any Okta admin make this request? Are there any other permissions required besides the FF enabled?

> As an Okta admin, make a `POST /oauth2/v1/clients/${yourServiceAppId}/roles` request to your org with the following required parameters to assign an admin role:

Reviewer: Good point @vanngo-okta.
Only SuperAdmin can assign roles.

> As an Okta admin, make a `POST /oauth2/v1/clients/${yourServiceAppId}/roles` request to your org with the following required parameters to assign an admin role:

Reviewer 2: Thanks for letting us know. I'll update [PR 4293](https://github.com/okta/okta-developer-docs/pull/4293) for the Org2Org use case with this info too.

| Parameter |  Description/Value   |
| --------- |  ------------- |

> | `yourServiceAppId`  |  Specify the `client_id` value from the previous response when the service app was created. In the following role assignment example, the `${yourServiceAppId}` variable name is used instead of `client_id`.|

Reviewer 2: ```suggestion
| `yourServiceAppId`  |  Specify the client ID value from the response when the [service app was created](#create-a-service-app-integration).|
```
The `client_id` variable is not referenced earlier. I suggest to link to the section where the service app was created, but you can change the wording or the link to something that is more appropriate.

> | `type`  |  Specify the admin role to assign to the service app. |

Reviewer 3: is there a long list of options for the value of `type`? Should we include what's supported here if the value list short and perhaps a link to a longer list if it's a long list??

Author: Good point! I'll add a link to this parameter and the two that follow.
